### PR TITLE
Packer accepts bytearray objects

### LINKED ISSUE #224: can't serialize bytearray
If value = bytearray(b'\x01\x87h\x05')

then packb(value, use_bin_type=True)

returns the error: can't serialize bytearray(b'\x01\x87h\x05')

Is this a known problem?

### PR DESCRIPTION
**Pull request by jfolz**
Fixes #224.

### CODE REVIEW

**Comment by jfolz** [comment]
Tiny niggle: It now says "Str is too large" instead of "Bytes is too large" in 2.7 if the object exceeds `ITEM_LIMIT`.

File: msgpack/_packer.pyx [hunk 1]
```diff
@@ -174,10 +184,11 @@ cdef class Packer(object):
                 else:
                    dval = o
                    ret = msgpack_pack_double(&self.pk, dval)
-            elif PyBytes_CheckExact(o) if strict_types else PyBytes_Check(o):
+            elif PyBytesLike_CheckExact(o) if strict_types else PyBytesLike_Check(o):
                 L = len(o)
                 if L > ITEM_LIMIT:
-                    raise PackValueError("bytes is too large")
+                    raise PackValueError("%s is too large"
+                                         % type(o).__name__.capitalize())
```
Comment (methane) [inline msgpack/_packer.pyx]: I don't like `capitalize()`.
How about `"%s length is too large" % type(o).__name__`?

Comment (jfolz) [inline msgpack/_packer.pyx]: No problem. `capitalize` is just there because it said `Bytes` instead of `bytes` ;)

**Comment by methane** [comment]
thanks.

**Pull request merged**